### ISSUE TITLE
Entering blockchain addresses manually isn't efficient and is actually impossible.

### ISSUE THREAD

**Issue by JezSan** [rotki/rotki#1539]
## Abstract

All Bitcoin addresses are generated from the xpub seed address.   Since on bitcoin it starts 2 new addresses for every transaction you do, it becomes impossible to manually track them and you NEED to be able to import an xpub seed and for it to find all the addresses itself.   I've been using bitcoin since 2013, and I've now got thousands of addresses.  

## Motivation

In this section please describe how you propose to address the problem you described earlier. Try to include example use cases that showcase the value of your proposal.

## Specification

If the feature is techical in nature please write as detailed as possible a specification of what needs to be built.

**Comment by JezSan** [comment]
Similarly, for Ethereum addresses, its not quite as bad, because the wallets don't automatically generate new addresses with each transaction, but the problem is still there and it'd be great if it had a way of generating all of your ethereum addresses somehow from a seed, but don't know if th equivalent of xpub is available in ethereum like it is in btc.  might need a safe offline tool to enter your seed words, and it to find all the addresses and export them.

**Comment by LefterisJP** [comment]
Hey @JezSan 

> but the problem is still there and it'd be great if it had a way of generating all of your ethereum addresses somehow from a seed, but don't know if th equivalent of xpub is available in ethereum like it is in btc

There may be some wallet specific ways to do it in ethereum but nothing like xpub in BTC afaik.

We are gonna attempt to do some wallet specific imports like from metamask.

> ince on bitcoin it starts 2 new addresses for every transaction you do, it becomes impossible to manually track them and you NEED to be able to import an xpub seed 

This is coming the coming week. It's implemented in develop at the moment and waiting to be included in the upcoming release: https://github.com/rotki/rotki/issues/808

Is that good enough for you to close the issue?

**Comment by JezSan** [comment]
that's a start ;-)

but we need a way to automatically import Ethereum addresses somehow.  even an import from another wallet won't work as they don't know all the addresses on a ledger wallet, and all they let you do is bring up a menu of your addresses and you choose the one you're going to work with, so at no time do you have all of the addresses in your ledger wallet available.

**Comment by LefterisJP** [comment]
Yes but we can't do anything else if it's not supported in the ethereum protocol. What we could do is allow you to import multiple addresses at once. So you paste a comma separated list of addresses and we add all of them. Would that work?

I am trying to manage the issues in the repo so I want to know what to do with this one :)

**Comment by JezSan** [comment]
close it if u feel its appropriate, but please do try and consider a way to deal with this... that there needs to be a more efficient way of getting multiple Ethereum addresses into it.. somehow.  those who have lots of addresses on the same ledger - all connected via the same seed - must surely have a more efficient way of entering the than one by one with cut and paste. could be hundreds or thousands for some.

**Comment by LefterisJP** [comment]
I don't own a ledger. So I can't tell you. But if it's a BIP39 then it may be possible. But you would need to also give rotki the seed and the password so that all addresses can be generated. Which is not as nice as xpub ... but at least you can see our code and be sure that we don't use it anywhere else.

My point with closing was that you were not asking for something specific and the initial thing you asked about bitcoin xpubs is already done and will be released this week.

But for BIP39 + ethereum we can probably schedule something. I will keep the issue for this. But will rewrite your original post to make this clear.

**Comment by LefterisJP** [comment]
So I think the way to do this would be to use the ledger connection directly.

Some resources here: https://twitter.com/BTChip/status/1420764191208525836

And a small package written doing exactly what we would need here: https://twitter.com/onybose/status/1420833705392803846
